applications/new_schema.py

```diff
@@ -138,9 +136,9 @@ class Query:
         "\n* A value passed is not a valid status",
     )
 
-    @login_required
-    @superuser_required
-    # TODO: Should check if the user has permissions to access these objects
+    @view_permission_required(
+        BerthApplication, BerthSwitch, Harbor, BerthSwitchReason, HarborChoice
```

Comment:
kinda related to the comment above. however, here you check also for `Harbor`, which is an open model with no view permissions required. should we maybe skip it from this list?

Reply:
And good catch here as well, I'll be sure to remove the public models from the `view_permissions` tests.

Reply:
On top of it: `BerthSwitchReason` belongs to the `applications` app, so as per the `User roles` doc it has the same permissions as `BerthApplication`, but it doesn't make much sense to "hide" it since it's only general information. Should we change the perms so that every role can `"view"` it?

Reply:
good point: `BerthSwitchReason` was not properly mentioned anywhere, but it is a public model. every role should have `"view"` permissions, but also since it is public we should not check if the user has permissions to it in the schema files.